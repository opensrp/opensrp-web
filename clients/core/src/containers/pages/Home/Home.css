.welcome-box {
    margin-top: 75px;
}
