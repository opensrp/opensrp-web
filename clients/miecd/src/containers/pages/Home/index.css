.welcome-box {
  margin-top: 75px;
}

.home-main .components-list {
  display: flex;
  padding-left: 25px;
  position: relative;
  flex-direction: row;
  height: 400px;
}

.nav {
  background-color: rgb(246, 254, 255);
  max-width: 25%;
  min-width: 140px;
  flex: 1;
  font-family: 'Roboto', 'Helvetica';
}

.main-container {
  display: flex;
  font-family: 'Roboto', 'Helvetica';
  font-size: 16px;
  background: white;
}

.main-container .content {
  height: 100%;
  position: relative;
  background: #f8fafe;
}

.home-main {
  padding: 50px 50px;
  background: white;
  height: 100%;
  position: relative;
}

.home-main .welcome-text {
  text-align: center;
  padding: 50px;
}

.home-main .welcome-text h1 {
  color: #1b314b;
  font-size: 32px;
  font-weight: 500;
  letter-spacing: -0.33px;
  line-height: 38px;
  text-align: center;
}

.home-main .button-style {
  border-radius: 24px;
  border: 1px solid #4d54b4;
  background: initial;
  border-radius: 24px;
  width: 100%;
  color: #4d54b4;
}

.home-main .button-style.default {
  border-radius: 24px;
  width: 100%;
  color: #ffffff;
  font-family: Roboto;
  letter-spacing: 0.1px;
  color: #4d54b4;
  height: 40px;
}

.home-main .button-style.default:hover {
  color: #ffffff;
  background: #4d54b4;
}

.home-main .home-sub-containers {
  background-color: #f7f6fb;
  border: 1px solid #dadce0;
  border-radius: 18px;
  box-shadow: 0 1px 3px 0 #d7d7e0;
  padding: 10px;
  height: 100%;
  width: 100%;
}

.sub-container-heading {
  border: none;
  font-size: 22px;
  text-align: left;
}

.home-main .sub-container-message {
  font-size: 1rem;
  margin: 5%;
  height: 100%;
}

.home-main .spacer {
  height: 100%;
  flex: 1 1 auto;
  margin: 25px;
  position: relative;
}

h4 p {
  text-align: left;
}

.home-main #cont-size {
  height: 10%;
  position: relative;
  top: 35%;
  color: #637381;
}

.home-sub-containers hr {
  margin-left: -10px;
  margin-right: -10px;
}

.home-main #cont-size h1 {
  color: #001429;
  font-size: 24px;
  font-weight: 500;
  letter-spacing: -0.27px;
  line-height: 29px;
  text-align: left;
}

.home-main #cont-sized button:focus {
  outline: none;
}

.home-main #cont-size h4 {
  color: #637381;
  font-size: 18px;
  font-weight: 500;
  letter-spacing: 0.1px;
  line-height: 22px;
}

.home-main #cont-sized {
  top: 50%;
  background: initial;
  position: relative;
  text-align: center;
}

@media screen and (max-width: 1100px) {
  .home-main {
    height: auto;
  }
  .home-main .components-list {
    flex-direction: column;
    height: auto;
    padding: 0;
  }
  .home-main .spacer img {
    left: 40% !important;
  }
  .home-main .spacer {
    width: 100%;
    height: 400px;
    margin: 60px 20px;
  }
}

img#womanchild1 {
  position: absolute;
  top: -21.5%;
  left: 36%;
}

img#womanchild2 {
  position: absolute;
  top: -22.4%;
  left: 40%;
}

img#womanchild3 {
  position: absolute;
  top: -20.4%;
  left: 35%;
}
