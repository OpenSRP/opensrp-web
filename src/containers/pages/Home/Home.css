.welcome-box {
  margin-top: 75px;
}

.components-list {
  display: flex;
  height: 40vh;
}

.nav {
  background-color: rgb(246, 254, 255);
  max-width: 25%;
  min-width: 140px;
  flex: 1;
  font-family: 'Trebuchet MS', 'Helvetica';
}

.main-container {
  display: flex;
  font-family: 'Trebuchet MS', 'Helvetica';
  font-size: 16px;
}

.home-main {
  padding: 50px 50px;
  width: 85vw;
}

.welcome-text {
  text-align: left;
  padding: 50px;
}

.button-style {
  padding: 10px 20px;
  border: 1px solid rgb(55, 175, 255);
  width: 75%;
}

.home-sub-containers {
  background-color: rgb(255, 255, 255);
  border-radius: 5px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  padding: 10px;
  height: 100%;
  width: 100%;
}

.sub-container-heading {
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  font-size: 1.2rem;
  text-align: left;
  height: 16%;
}

.sub-container-message {
  font-size: 1rem;
  margin: 10%;
  height: 84%;
}

.spacer {
  height: 300px;
  width: 33.3%;
  margin: 25px;
}

h4 p {
  text-align: left;
}

#cont-size {
  height: 50%;
}

#cont-sized {
  height: 50%;
  text-align: center;
}
