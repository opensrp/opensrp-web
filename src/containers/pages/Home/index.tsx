// this is the home page component
import * as React from 'react';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import womanchild1 from '../../../assets/images/womanchild1.png';
import womanchild2 from '../../../assets/images/womanchild2.png';
import womanchild3 from '../../../assets/images/womanchild3.png';
import { LOGFACE_URL, NBC_AND_PNC_URL, NUTRITION_URL, PREGNANCY_URL } from '../../../constants';
import { LogFace } from '../../LogFace';
import './index.css';

class Home extends React.Component<{}, {}> {
  public render() {
    return (
      <div className="home-main">
        <div className="welcome-text">
          <h1>Welcome to the MIECD dashboard</h1>
        </div>
        <div className="components-list">
          <div className="spacer">
            <img src={womanchild1} id="womanchild1" />
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>Pregnancy</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  <Link to={PREGNANCY_URL}>
                    <button className="button-style default">View</button>
                  </Link>
                </div>
              </div>
            </div>
          </div>
          <div className="spacer">
            <img src={womanchild2} id="womanchild2" />
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>NBC & PNC</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  <Link to={NBC_AND_PNC_URL}>
                    <button className="button-style default">View</button>
                  </Link>
                </div>
              </div>
            </div>
          </div>
          <div className="spacer">
            <img src={womanchild3} id="womanchild3" />
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>Nutrition</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  <Link to={NUTRITION_URL}>
                    <button className="button-style default">View</button>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
