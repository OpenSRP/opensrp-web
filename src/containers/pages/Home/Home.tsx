// this is the home page component
import * as React from 'react';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { LOGFACE_URL } from '../../../constants';
import { LogFace } from '../../LogFace';
import './Home.css';

class Home extends React.Component<{}, {}> {
  public render() {
    return (
      <div className="home-main">
        <div className="welcome-text">
          <h1>Welcome to the MIECD dashboard</h1>
        </div>
        <div className="components-list">
          <div className="spacer">
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>Pregnancy</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  <Link to={'/pregnancy'}>
                    <button className="button-style default">View</button>
                  </Link>
                </div>
              </div>
            </div>
          </div>
          <div className="spacer">
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>NBC & PNC</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  
                    <button className="button-style">View</button>
                  
                </div>
              </div>
            </div>
          </div>
          <div className="spacer">
            <div className="home-sub-containers">
              {/* <div className="sub-container-heading">
                <p>Log face</p>
              </div> */}
              <div className="sub-container-message">
                <div id="cont-size">
                  <h1>Nutrition</h1>
                </div>
                <div id="cont-size">
                  <h4>View the latest message updates about your patients.</h4>
                </div>
                <div id="cont-sized">
                  
                    <button className="button-style">View</button>
                  
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
