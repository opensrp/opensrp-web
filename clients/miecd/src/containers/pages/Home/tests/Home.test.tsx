import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import Home from '..';

const history = createBrowserHistory();

describe('containers/pages/Home', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('renders without crashing', () => {
        shallow(
            <Router history={history}>
                <Home />
            </Router>,
        );
    });

    it('renders Home correctly', () => {
        const wrapper = mount(
            <Router history={history}>
                <Home />
            </Router>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
