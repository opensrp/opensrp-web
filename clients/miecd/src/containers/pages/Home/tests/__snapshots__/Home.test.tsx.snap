// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Home renders Home correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Home>
    <div
      className="home-main"
    >
      <div
        className="welcome-text"
      >
        <h1>
          Welcome to the MIECD dashboard
        </h1>
      </div>
      <div
        className="components-list"
      >
        <div
          className="spacer"
        >
          <img
            alt="woman"
            id="womanchild1"
            src="test-file-stub"
          />
          <div
            className="home-sub-containers"
          >
            <div
              className="sub-container-message"
            >
              <div
                id="cont-size"
              >
                <h1>
                  Pregnancy
                </h1>
              </div>
              <div
                id="cont-size"
              >
                <h4>
                  View the latest message updates about your patients.
                </h4>
              </div>
              <div
                id="cont-sized"
              >
                <Link
                  to="/pregnancy"
                >
                  <LinkAnchor
                    href="/pregnancy"
                    navigate={[Function]}
                  >
                    <a
                      href="/pregnancy"
                      onClick={[Function]}
                    >
                      <button
                        className="button-style default"
                      >
                        View
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </div>
        </div>
        <div
          className="spacer"
        >
          <img
            alt="woman"
            id="womanchild2"
            src="test-file-stub"
          />
          <div
            className="home-sub-containers"
          >
            <div
              className="sub-container-message"
            >
              <div
                id="cont-size"
              >
                <h1>
                  NBC & PNC
                </h1>
              </div>
              <div
                id="cont-size"
              >
                <h4>
                  View the latest message updates about your patients.
                </h4>
              </div>
              <div
                id="cont-sized"
              >
                <Link
                  to="/nbc_and_pnc"
                >
                  <LinkAnchor
                    href="/nbc_and_pnc"
                    navigate={[Function]}
                  >
                    <a
                      href="/nbc_and_pnc"
                      onClick={[Function]}
                    >
                      <button
                        className="button-style default"
                      >
                        View
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </div>
        </div>
        <div
          className="spacer"
        >
          <img
            alt="woman"
            id="womanchild3"
            src="test-file-stub"
          />
          <div
            className="home-sub-containers"
          >
            <div
              className="sub-container-message"
            >
              <div
                id="cont-size"
              >
                <h1>
                  Nutrition
                </h1>
              </div>
              <div
                id="cont-size"
              >
                <h4>
                  View the latest message updates about your patients.
                </h4>
              </div>
              <div
                id="cont-sized"
              >
                <Link
                  to="/nutrition"
                >
                  <LinkAnchor
                    href="/nutrition"
                    navigate={[Function]}
                  >
                    <a
                      href="/nutrition"
                      onClick={[Function]}
                    >
                      <button
                        className="button-style default"
                      >
                        View
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Home>
</Router>
`;
